src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java

```diff
@@ -172,7 +177,7 @@
         if (problemStatement == null || problemStatement.isEmpty()) {
             return tasks;
         }
-        var matcher = taskPatternForProblemStatementMarkdown.matcher(problemStatement);
+        var matcher = TASK_PATTERN.matcher(problemStatement);
```

Comment:
🤔 I don't really see a ways to improve the regex, has someone else an idea?

Reply:
@Strohgelaender You could delete part of the pattern string and see if CodeQL is happy then. Then we'd know what part is the problem.

A non-greedy capture might wor, otherwise you could try
`\\[task]\\[(?<name>[^\\[\\]]+)]\\((?<tests>\w[\w_]*(?:,\s*[\w_]*)*)\\)`
It should be more precise. But again, first we need to know which part is the problem. Cut it out and if  CodeQL resolves this issue, that was it.